### PR TITLE
Adding `Fallback` on election failure

### PR DESCRIPTION
**Pull request by georgesdib**
Companion to paritytech/substrate#10988

Use the newly introduced `BoundedOnchainExecution `, and `UnboundedOnchainExecution ` for Genesis.

@kianenigma @emostov

### CODE REVIEW

**Review by kianenigma — CHANGES_REQUESTED** [review]
`type Fallback` in our production networks should remain `NoFallback` since we don't quite know what are some good bounds. 

As for the `GovernanceFallback` as well, since this transaction can impose its own limits, I think actually for now we should set it to `UnboundedExecution`, because we don't know any 'reasonable static bounds' that we can assign to any `BoundedExecution` type. 

The missing step to figure all of this out is to create an accurate weight function for all of this `Solver` implementations, will make an issue for it.

**Comment by georgesdib** [comment]
> `type Fallback` in our production networks should remain `NoFallback` since we don't quite know what are some good bounds.
> 
> As for the `GovernanceFallback` as well, since this transaction can impose its own limits, I think actually for now we should set it to `UnboundedExecution`, because we don't know any 'reasonable static bounds' that we can assign to any `BoundedExecution` type.
> 
> The missing step to figure all of this out is to create an accurate weight function for all of this `Solver` implementations, will make an issue for it.

I have reverted the changes including for Westend.

**Review by kianenigma — APPROVED** [review]

File: runtime/test-runtime/src/lib.rs [hunk 1]
```diff
@@ -342,11 +337,15 @@ impl pallet_staking::Config for Runtime {
...
-	type VoterList = pallet_staking::UseNominatorsAndValidatorsMap<Runtime>;
+	type ElectionProvider = frame_election_provider_support::onchain::UnboundedExecution<
+		runtime_common::elections::OnChainSeqPhragmen<Self, Staking>,
+	>;
+	type GenesisElectionProvider = frame_election_provider_support::onchain::UnboundedExecution<
+		runtime_common::elections::OnChainSeqPhragmen<Self, Staking>,
+	>;
+	// Use the nominator map to iter voter AND no-ops for all SortedListProvider hooks. The migration
+	// to bags-list is a no-op, but the storage version will be updated.
+	type VoterList = pallet_staking::UseNominatorsMap<Runtime>;
```
Comment (kianenigma) [inline runtime/test-runtime/src/lib.rs:348]:
```suggestion
	type VoterList = pallet_staking::UseNominatorsAndValidatorsMap<Runtime>;
```

**Comment by kianenigma** [comment]
@georgesdib thanks for making the bot work with your fork, helps with the automated process.

**Comment by kianenigma** [comment]
Must look into why and how this caused benchmarks to break. 

Alternatively, we can merge this to keep the normal build of master working.

**Comment by georgesdib** [comment]
> Must look into why and how this caused benchmarks to break.
> 
> Alternatively, we can merge this to keep the normal build of master working.

You think it's caused by the changes here? I tried to go back in time and see merged PRs from 2 days ago, and they seem to fail the same thing.

**Comment by kianenigma** [comment]
hope I am not breaking anything for anyone, merging this manually to keep the master working.

**Pull request merged**